R/dependency.R

```diff
@@ -0,0 +1,104 @@
+#' Add an external JavaScript asset as a dygraph dependency
+#'
+#' @param dygraph Dygraph to add dependency to
+#' @param type Type of dependency (valid options are: 'plugins', 'plotter' or 'dataHandler')
+#' @param name Name of dependency
+#' @param path Path to dependency JavaScript file
+#' @param version Dependency version (e.g. version of package which provides the dependency)
+#' @param all.files Whether all files under the path directory are dependency files
+#'
+#' @return A dygraph with the specified dependency added.
+#'
+#' @importFrom htmltools htmlDependency
+#'
+#' @export
+dyDependency <- function(dygraph, type, name, path, version = "1.0", all.files = FALSE) {
+  # create an html dependency for the js file
+  path <- normalizePath(path)
+  assetDependency <- htmlDependency(depName(type, name),
+                                    version,
+                                    src = dirname(path),
+                                    script = basename(path),
+                                    all_files = all.files)
+
+  # add javascript to the dependencies
+  if (is.null(dygraph$dependencies)) {
+    dygraph$dependencies <- list()
+  }
+  dygraph$dependencies[[length(dygraph$dependencies) + 1]] <- assetDependency
+
+  # return dygraph
+  dygraph
+}
+
+# Make up a htmlDependency name based on asset type and name
+depName <- function(type, name) {
```

Comment:
I think once we change the interface of `dyDependency` to take an html dependency directly this function can go away.